### ISSUE TITLE
EnC - unknown error when making some edits

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#52147]
**Version Used**: 16.9.2 and above

**Steps to Reproduce**:

1. Have this code:
```
using System;

namespace ConsoleApp277
{
    class Program
    {
        static void Main(string[] args)
        {
            // comment1
            Console.WriteLine("asdf");
            // comment2
        }

        void M()
        {
            // comment3
        }
    }

    public class R
    {
        // comment4
    }
}
```

2. Put a breakpoint on Console.WriteLine
3. Press F5
4. When broken, you can successfully delete any of the comments and continue (F10 or F5) except comment4

**Expected Behavior**:

Should be able to delete comment4

**Actual Behavior**:
	
Error, the source file has changed.

On legacy projects this reports a diagnostic, in SDK-style it shows a messagebox.

**Comment by tmat** [comment]
What diagnostic is reported?

**Comment by davidwengier** [comment]
This is the error.
![image](https://user-images.githubusercontent.com/754264/112544471-26b34780-8e0b-11eb-9800-4c2e4fc3aaf8.png)

I saw a diagnostic reported in the error list for a legacy project last night on my laptop, but thats not reproing for me now. I might have done something wrong and made a genuine rude edit accidentally.

**Comment by davidwengier** [comment]
Looks like its fixed in main :)
